docs/src/assets/js/themes.js

```diff
@@ -1,54 +1,66 @@
 /* theme toggle buttons */
-(function() {
-    var enableToggle = function(btn) {
+(function () {
+    var enableToggle = function (btn) {
         btn.setAttribute("aria-pressed", "true");
     }
-
-    var disableToggle = function(btn) {
-        btn.setAttribute("aria-pressed", "false");
+    var disableToggle = function (btns) {
+        btns.forEach(btn => btn.setAttribute("aria-pressed", "false"));
+    }
+    var setTheme = function (theme) {
+        if (theme === "system") {
+            theme = window.matchMedia('(prefers-color-scheme: dark)').matches ? "dark" : "light";
+            document.documentElement.setAttribute('data-theme', theme);
+            window.localStorage.setItem("theme", "system");
+        } else {
+            document.documentElement.setAttribute('data-theme', theme);
+            window.localStorage.setItem("theme", theme);
+        }
     }
```

Comment:
[Refactor] I think this function can be simplified, since the `setAttribute` call happens no matter what, and the `setItem` call is always done with the original value of `theme`?

```suggestion
    var setTheme = function (theme) {
        window.localStorage.setItem("theme", theme);
        if (theme === "system") {
            theme = window.matchMedia('(prefers-color-scheme: dark)').matches ? "dark" : "light";
        }
        document.documentElement.setAttribute('data-theme', theme);
    }
```

Reply:
refactored